### DISCUSSION TITLE
Upgrading from 19.4.x to 20: undefined function idn_to_ascii (with php 8.3)

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [OpenMage/magento-lts#3930] [Questions & Answers] [answered]
Call to undefined function idn_to_ascii() in /app/code/core/Mage/Core/Helper/Url.php:169

https://github.com/OpenMage/magento-lts/pull/1505

**Comment by alexh-swdev** [comment]
Maybe a problem with your config? It's a standard PHP function: https://www.php.net/manual/de/function.idn-to-ascii.php

**Reply by f1-outsourcing** [reply, marked as answer]
Yes indeed this one was missing
php83-php-intl-8.3.3-1.el9.remi.x86_64.rpm

**Reply by pquerner** [reply]
Thats also one of the requirements. Composer should have bugged you when its missing. Or did you install on some other ways?

(from the readme)

`PHP extension intl since 1.9.4.19 & 20.0.17`

**Reply by f1-outsourcing** [reply]
I am downloading zip, maybe it should be in this install procedure. If I remember correctly it gives some recommendations not? Around this time when it mentions what directories should be writeable.

**Reply by pquerner** [reply]
Yes, it should. It was also part of the original PR https://github.com/OpenMage/magento-lts/commit/59550afc34169188865cac19f0d5eebb28fa6a28

**Reply by fballiano** [reply]
it's in the readme https://github.com/openmage/magento-lts/?tab=readme-ov-file#requirements

**Reply by luigifab** [reply]
I think you should take the time to update your installation with composer. You will be happy for the next upgrade!

**Reply by f1-outsourcing** [reply]
> I think you should take the time to update your installation with composer. You will be happy for the next upgrade!

Yes you are maybe right, but I have such crappy experience with this composer, I don't get why it consumes so much memory sometimes and downloads development crap like xdebug or so. It seems highly inefficient to me. Last time I used it on a helloworld example it seemed to be busy with unrelated things.
